inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp

```diff
@@ -0,0 +1,164 @@
+        const auto transpose = std::make_shared<opset1::Transpose>(constant, transposeConstant);
+        return transpose;
+    }
+
+    std::shared_ptr<ngraph::Node> makeConstant(const ngraph::element::Type precision, const ngraph::Shape shape, const std::vector<float> low) {
+        return std::make_shared<opset1::Constant>(precision, shape, low);;
+    }
+
+    std::shared_ptr<ngraph::Function> get(
+        ngraph::element::Type precision,
+        ngraph::Shape inputShape,
+        const bool inputLowConst,
+        const bool inpuHighConst,
+        const bool outputLowConst,
+        const bool outputHighConst) {
+        const auto input = std::make_shared<ngraph::opset1::Parameter>(precision, inputShape);
+        input->set_friendly_name("input");
+
+        const auto constantPresition = element::f32;
+        const auto constantShape = Shape{ 1, 1, 1, 1 };
+        const std::vector<float> low = { 0.f };
+        const std::vector<float> high = { 1.f };
+
+        const auto inputLow = inputLowConst ?
+            makeConstant(constantPresition, constantShape, low) :
+            makeTranspose(constantPresition, constantShape, low);
+
+        const auto inputHigh = inputLowConst ?
+            makeConstant(constantPresition, constantShape, high) :
+            makeTranspose(constantPresition, constantShape, high);
+
+        const auto outputLow = outputLowConst ?
+            makeConstant(constantPresition, constantShape, low) :
+            makeTranspose(constantPresition, constantShape, low);
+
+        const auto outputHigh = outputHighConst ?
+            makeConstant(constantPresition, constantShape, high) :
+            makeTranspose(constantPresition, constantShape, high);
+
+        const auto levels = 256ul;
+
+        auto fakeQuantize = std::make_shared<ngraph::opset1::FakeQuantize>(input, inputLow, inputHigh, outputLow, outputHigh, levels);
+        fakeQuantize->set_friendly_name("fakeQuantize");
+
+        ngraph::ResultVector results{ std::make_shared<ngraph::opset1::Result>(fakeQuantize) };
+        return std::make_shared<ngraph::Function>(results, ngraph::ParameterVector{ input }, "FakeQuantizeWithDynamicIntervalsTransformation");
+    }
+};
+
+TEST_P(FakeQuantizeWithDynamicIntervalsTransformation, CompareFunctions) {
+    actualFunction->validate_nodes_and_infer_types();
+    auto res = compare_functions(referenceFunction, actualFunction, true, true, true);
+    ASSERT_TRUE(res.first) << res.second;
+}
+
+const std::vector<ngraph::element::Type> precisions = {
+    ngraph::element::f32,
+    ngraph::element::i32,
+    ngraph::element::f16
+};
+
+const std::vector<FakeQuantizeWithDynamicIntervalsTransformationTestValues> fakeQuantizeTransformationTestValues = {
+    { LayerTransformation::createParamsU8I8(), false, false, false, true },
+    { LayerTransformation::createParamsU8I8(), true, false, false, false }
```

Comment:
Don't we need a case where all inputs are dynamic which is most common I suppose?

Reply:
thanks, was added